packages/twenty-front/src/modules/ui/layout/modal/components/__stories__/ConfirmationModal.stories.tsx

```diff
@@ -29,5 +60,117 @@ export const InputConfirmation: Story = {
+    title: 'Escape Key Test',
+    subtitle: 'This modal should close when pressing the Escape key.',
+    confirmButtonText: 'Confirm',
+    onClose: closeMock,
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    await canvas.findByText('Escape Key Test');
+
+    closeMock.mockClear();
+
+    await userEvent.keyboard('{Escape}');
+
+    expect(closeMock).toHaveBeenCalledTimes(1);
+  },
+};
+
+export const CloseOnClickOutside: Story = {
+  args: {
+    modalId: 'confirmation-modal',
+    title: 'Click Outside Test',
+    subtitle: 'This modal should close when clicking outside of it.',
+    confirmButtonText: 'Confirm',
+    onClose: closeMock,
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    await canvas.findByText('Click Outside Test');
+
+    closeMock.mockClear();
+
+    const backdrop = document.querySelector('.modal-backdrop') as HTMLElement;
+    await userEvent.click(backdrop);
+
+    expect(closeMock).toHaveBeenCalledTimes(1);
+  },
+};
+
+export const ConfirmWithEnterKey: Story = {
+  args: {
+    modalId: 'confirmation-modal',
+    title: 'Enter Key Test',
+    subtitle: 'This modal should confirm when pressing the Enter key.',
+    confirmButtonText: 'Confirm',
+    onConfirmClick: fn(),
+  },
+  play: async ({ canvasElement, args }) => {
+    const canvas = within(canvasElement);
+
+    await canvas.findByText('Enter Key Test');
+
+    const confirmMock = args.onConfirmClick as ReturnType<typeof fn>;
+    confirmMock.mockClear();
+
+    await userEvent.keyboard('{Enter}');
+
+    expect(confirmMock).toHaveBeenCalledTimes(1);
+  },
+};
+
+export const CancelButtonClick: Story = {
+  args: {
+    modalId: 'confirmation-modal',
+    title: 'Cancel Button Test',
+    subtitle: 'Clicking the cancel button should close the modal',
+    confirmButtonText: 'Confirm',
+    onClose: fn(),
+  },
+  play: async ({ canvasElement, args }) => {
+    const canvas = within(canvasElement);
+
+    await canvas.findByText('Cancel Button Test');
+
+    const closeMock = args.onClose as ReturnType<typeof fn>;
+    closeMock.mockClear();
+
+    const cancelButton = canvas.getByTestId('confirmation-modal-cancel-button');
+    await userEvent.click(cancelButton);
+
+    expect(closeMock).toHaveBeenCalledTimes(1);
+  },
+};
+
+export const ConfirmButtonClick: Story = {
+  args: {
+    modalId: 'confirmation-modal',
+    title: 'Confirm Button Test',
+    subtitle: 'Clicking the confirm button should trigger the confirm action',
+    confirmButtonText: 'Confirm',
+    onConfirmClick: fn(),
+  },
+  play: async ({ canvasElement, args }) => {
+    const canvas = within(canvasElement);
+
+    await canvas.findByText('Confirm Button Test');
+
+    const confirmMock = args.onConfirmClick as ReturnType<typeof fn>;
+    confirmMock.mockClear();
+
+    const confirmButton = canvas.getByTestId(
+      'confirmation-modal-confirm-button',
+    );
```

Comment:
We aren't forced to rely on the data-testid

```suggestion
    const confirmButton = canvas.getByRole('button', {
      name: /Confirm/,
    });
```